vowpalwabbit/learner.cc

```diff
@@ -186,10 +186,10 @@ class multi_example_handler
       _context.template process<multi_ex, learn_multi_ex>(ec_seq);
       ec_seq.clear();
     }
-
-    // Send out the end-of-pass notification after doing learning
-    if (ec->end_pass)
+    else if (ec->end_pass)
```

Comment:
`else if` branch will be true when we call `on_example` again with the new `end_pass` example object after it is `pop`ed from the `ready_example` queue